model/trigger/task.go

```diff
@@ -363,3 +368,131 @@ func (t *taskTriggers) taskRegression(sub *event.Subscription) (*notification.No
 
 	return n, errors.Wrap(rec.Insert(), "failed to process regression trigger")
 }
+
+func isFailedTaskStatus(status string) bool {
+	return status == evergreen.TaskFailed || status == evergreen.TaskSystemFailed || status == evergreen.TaskTestTimedOut
+}
+
+func isTestRegression(oldStatus, newStatus string) bool {
+	switch oldStatus {
+	case evergreen.TestSkippedStatus, evergreen.TestSucceededStatus,
+		evergreen.TestSilentlyFailedStatus:
+		if newStatus == evergreen.TestFailedStatus {
+			return true
+		}
+	}
+
+	return false
+}
+
+func isTaskRegression(oldStatus, newStatus string) bool {
+	switch oldStatus {
+	case evergreen.TaskFailed:
+		if newStatus == evergreen.TaskSystemFailed || newStatus == evergreen.TaskTestTimedOut {
+			return true
+		}
+
+	case evergreen.TaskSystemFailed:
+		if newStatus == evergreen.TaskFailed || newStatus == evergreen.TaskTestTimedOut {
+			return true
+		}
+
+	case evergreen.TaskSucceeded:
+		return isFailedTaskStatus(newStatus)
+	}
+
+	return false
+}
+
+func (t *taskTriggers) shouldIncludeTest(previousTask *task.Task, test *testresult.TestResult) (bool, error) {
+	if test.Status != evergreen.TestFailedStatus {
+		return false, nil
+	}
+	record, err := alertrecord.FindByLastTaskRegressionByTest(test.TestFile, t.task.DisplayName, t.task.BuildVariant, t.task.Project, t.task.RevisionOrderNumber)
+	if err != nil {
+		return false, errors.Wrap(err, "Failed to fetch alert record")
+	}
+	if record != nil {
+		if record.RevisionOrderNumber == t.task.RevisionOrderNumber || previousTask == nil {
+			return false, nil
+		}
+
+		oldTestResult, ok := t.oldTestResults[test.TestFile]
+		if !ok {
+			if test.Status != evergreen.TestFailedStatus {
+				return false, nil
+			}
+
+		} else if !isTestRegression(oldTestResult.Status, test.Status) {
+			return false, nil
+		}
+	}
+
+	err = alertrecord.InsertNewTaskRegressionByTestRecord(test.TestFile, t.task.DisplayName, t.task.BuildVariant, t.task.Project, t.task.RevisionOrderNumber)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to save alert record")
+	}
+	return true, nil
+}
+
+func (t *taskTriggers) taskRegressionByTest(sub *event.Subscription) (*notification.Notification, error) {
+	if t.task.Requester != evergreen.RepotrackerVersionRequester || !isFailedTaskStatus(t.task.Status) {
+		return nil, nil
+	}
+
+	previousCompleteTask, err := task.FindOne(task.ByBeforeRevisionWithStatuses(t.task.RevisionOrderNumber,
+		task.CompletedStatuses, t.task.BuildVariant, t.task.DisplayName, t.task.Project))
+	if err != nil {
+		return nil, errors.Wrap(err, "error fetching previous task")
+	}
+
+	tests, err := testresult.FindByTaskIDAndExecution(t.task.Id, t.task.Execution)
```

Comment:
Since FindOne already merges test results, you should just iterate through those in memory

Reply:
Fixed